src/daemons/linuxd/src/linuxd/mod.rs

```diff
@@ -86,103 +73,26 @@ impl LinuxDaemon {
-                    match gw_stdin_rx.recv() {
-                        Ok((_read_request, response_tx)) => {
-                            let mut response_buf: [u8; ReadResponse::BUFFER_SIZE] =
-                                [0u8; ReadResponse::BUFFER_SIZE];
-                            let num_read = match gw_stdin_stream.read(&mut response_buf) {
-                                Ok(n) => n,
-                                Err(e) => {
-                                    let reason: String =
-                                        format!("failed to read STDIN from gateway: {e:?}");
-                                    error!("{}", reason);
-                                    return Err(anyhow::anyhow!(reason));
-                                },
-                            };
-                            response_tx.send(ReadResponse::build(
-                                0.into(),
-                                num_read as c_ssize_t,
-                                response_buf,
-                            ))?;
-                        },
-                        Err(RecvError) => {
-                            // This may happen during shutdown.
-                            debug!("gateway STDIN channel disconnected");
-                            break Ok(());
-                        },
-                    }
-                }
-            });
-
-            let mut gw_stdout_stream = gateway_stream
-                .try_clone()
-                .map_err(|_| Error::new(ErrorCode::IoErr, "failed to clone stream"))?;
-            let _gw_stdout_thread: JoinHandle<Result<()>> = std::thread::spawn(move || {
-                loop {
-                    // Block waiting the user VM to request writing to stdout.
-                    match gw_stdout_rx.recv() {
-                        Ok(write_request) => {
-                            gw_stdout_stream
-                                .write_all(&write_request.buffer[..write_request.count as usize])?;
-
-                            // We don't need to send anything in response of the write, as the
-                            // writting thread has already moved on.
-                        },
-                        Err(RecvError) => {
-                            // This may happen during shutdown.
-                            debug!("gateway STDOUT channel disconnected");
-                            break Ok(());
-                        },
-                    }
-                }
-            });
-
-            (Some(gw_stdin_tx), Some(gw_stdout_tx))
-        } else {
-            (None, None)
-        };
-
+    pub fn run(self) -> Result<(), Error> {
         // Map keeping track of the worker threads associated the user VM.
         let mut worker_threads: VecDeque<WorkerThreadHandle> = VecDeque::new();
 
+        // Handle around the user VM.
+        let uvm_handle: UserVmHandle = UserVmHandle::new(0, self.uvm_stream, self.gateway_stream);
```

Comment:
Can you can create a file-cope (ie  `DEFAULT_CONNECTION_ID`) constant for the connection id `0`?